File: cpp/Array_Hashing/1.two-sum.cpp
#include "../std.hpp"
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> indice;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (indice.find(complement) != indice.end()) {
                return {indice[complement], i};
            }
            else {
                indice.insert({nums[i], i});
            }
        }
        return {};
    }
};
int main(){
    
}

File: TypeScript/Array_Hashing/1.two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const m = {};
    for (let i = 0; i < nums.length; ++i) {
        const compliment = target - nums[i];
        if (compliment in m) {
            return [m[compliment], i];
        }
        else {
            m[nums[i]] = i;
        }
    }
    return []
};
console.log(twoSum([3, 2, 4], 6));